allensdk/brain_observatory/behavior/behavior_project_cache/tables/util/metadata_parsers.py

```diff
@@ -0,0 +1,245 @@
+from typing import Union
+
+import pandas as pd
+
+#####################################################################
+#
+#                   VBO PROJECT CONSTANTS
+#
+#####################################################################
+
+
+SESSION_TYPES = [
+    "OPHYS_0_images_A_habituation",
+    "OPHYS_0_images_B_habituation",
+    "OPHYS_0_images_G_habituation",
+    "OPHYS_1_images_A",
+    "OPHYS_1_images_B",
+    "OPHYS_1_images_G",
+    "OPHYS_2_images_A_passive",
+    "OPHYS_2_images_B_passive",
+    "OPHYS_2_images_G_passive",
+    "OPHYS_3_images_A",
+    "OPHYS_3_images_B",
+    "OPHYS_3_images_G",
+    "OPHYS_4_images_A",
+    "OPHYS_4_images_B",
+    "OPHYS_4_images_H",
+    "OPHYS_5_images_A_passive",
+    "OPHYS_5_images_B_passive",
+    "OPHYS_5_images_H_passive",
+    "OPHYS_6_images_A",
+    "OPHYS_6_images_B",
+    "OPHYS_6_images_H",
+    "TRAINING_0_gratings_autorewards_15min",
+    "TRAINING_1_gratings",
+    "TRAINING_2_gratings_flashed",
+    "TRAINING_3_images_A_10uL_reward",
+    "TRAINING_3_images_B_10uL_reward",
+    "TRAINING_3_images_G_10uL_reward",
+    "TRAINING_4_images_A_handoff_lapsed",
+    "TRAINING_4_images_A_handoff_ready",
+    "TRAINING_4_images_A_training",
+    "TRAINING_4_images_B_training",
+    "TRAINING_4_images_G_training",
+    "TRAINING_5_images_A_epilogue",
+    "TRAINING_5_images_A_handoff_lapsed",
+    "TRAINING_5_images_A_handoff_ready",
+    "TRAINING_5_images_B_epilogue",
+    "TRAINING_5_images_B_handoff_lapsed",
+    "TRAINING_5_images_B_handoff_ready",
+    "TRAINING_5_images_G_epilogue",
+    "TRAINING_5_images_G_handoff_lapsed",
+    "TRAINING_5_images_G_handoff_ready",
+]
+
+PROJECT_CODES = [
+    "VisualBehavior",
+    "VisualBehaviorTask1B",
+    "VisualBehaviorMultiscope",
+    "VisualBehaviorMultiscope4areasx2d",
+]
+
+NUM_STRUCTURES_DICT = {
```

Comment:
these things really should be in the LIMS database rather than being hardcoded here

Reply:
Yep. Unfortunately this is what we've got to work with.